src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_IncrementalAnalyzer.cs

```diff
@@ -256,6 +267,26 @@ private void RaiseDiagnosticsRemovedIfRequiredForClosedOrResetDocument(TextDocum
             RaiseDiagnosticsRemovedForDocument(document.Id, stateSets);
         }
 
+        public async Task ActiveDocumentSwitchedAsync(TextDocument document, CancellationToken cancellationToken)
+        {
+            // When the analysis scope is set to 'ActiveFile' and the active document is switched,
+            // we retrigger analysis of newly active document.
+            if (SolutionCrawlerOptions.GetBackgroundAnalysisScope(document.Project) != BackgroundAnalysisScope.ActiveFile)
+            {
+                return;
```

Comment:
💡 A comment on this return describing the other locations where analysis is triggered for non-active-file mode would improve clarity. Or, if that's too long, a comment explaining why the others don't need to handle these events.